src/open_producten/producttypen/tests/api/test_prijs.py

```diff
@@ -0,0 +1,432 @@
+        response = APIClient().get(self.path)
+        self.assertEqual(response.status_code, 401)
+
+    def test_create_prijs_without_opties(self):
+        response = self.post(self.prijs_data)
+
+        self.assertEqual(response.status_code, 400)
+        self.assertEqual(
+            response.data,
+            {
+                "prijsopties": [
+                    ErrorDetail(
+                        string="Er is minimaal één optie vereist.",
+                        code="invalid",
+                    )
+                ]
+            },
+        )
+
+    def test_create_prijs_with_prijs_opties(self):
+        data = {
+            "actief_vanaf": datetime.date(2024, 1, 2),
+            "prijsopties": [{"bedrag": "74.99", "beschrijving": "spoed"}],
+            "product_type_id": self.product_type.id,
+        }
+
+        response = self.post(data)
+
+        self.assertEqual(response.status_code, 201)
+        self.assertEqual(Prijs.objects.count(), 1)
+        self.assertEqual(PrijsOptie.objects.count(), 1)
+        self.assertEqual(
+            self.product_type.prijzen.first().prijsopties.first().bedrag,
+            Decimal("74.99"),
+        )
+
+    def test_update_prijs_removing_all_opties(self):
+        prijs = self._create_prijs()
+        PrijsOptieFactory.create(prijs=prijs)
+        PrijsOptieFactory.create(prijs=prijs)
+
+        data = {
+            "actief_vanaf": prijs.actief_vanaf,
+            "product_type_id": self.product_type.id,
+            "prijsopties": [],
+        }
+
+        response = self.put(prijs.id, data)
+
+        self.assertEqual(response.status_code, 400)
+        self.assertEqual(
+            response.data,
+            {
+                "prijsopties": [
+                    ErrorDetail(
+                        string="Er is minimaal één optie vereist.",
+                        code="invalid",
+                    )
+                ]
+            },
+        )
+
+    def test_update_prijs_updating_and_removing_opties(self):
+        prijs = self._create_prijs()
+        optie_to_be_updated = PrijsOptieFactory.create(prijs=prijs)
+        PrijsOptieFactory.create(prijs=prijs)
+
+        data = {
+            "actief_vanaf": prijs.actief_vanaf,
+            "product_type_id": self.product_type.id,
+            "prijsopties": [
+                {
+                    "id": optie_to_be_updated.id,
+                    "bedrag": "20",
+                    "beschrijving": optie_to_be_updated.beschrijving,
+                }
+            ],
+        }
+
+        response = self.put(prijs.id, data)
+
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(Prijs.objects.count(), 1)
+        self.assertEqual(PrijsOptie.objects.count(), 1)
+        self.assertEqual(PrijsOptie.objects.first().bedrag, Decimal("20"))
+
+    def test_update_prijs_creating_and_deleting_opties(self):
+        prijs = self._create_prijs()
+        PrijsOptieFactory.create(prijs=prijs)
```

Comment:
`prijs_optie = PrijsOptieFactory.create(prijs=prijs)`

if you delete prijs_optie, its id will be set to -1 which might be useful. Not sure if it works with custom ids though